test/e2e/spworkflow/e2e_test.sh

```diff
@@ -0,0 +1,181 @@
+  set -e
+  cd ${workspace}
+  # build sp
+  git clone https://github.com/bnb-chain/greenfield-cmd.git
+  cd greenfield-cmd/
+  git checkout ${GREENFIELD_CMD_BRANCH}
+  make build
+  cd build/
+
+  # generate a keystore file to manage private key information
+  touch key.txt & echo ${TEST_ACCOUNT_PRIVATE_KEY} > key.txt
+  touch password.txt & echo "test_sp_function" > password.txt
+  ./gnfd-cmd --home ./ keystore generate --privKeyFile key.txt --passwordfile password.txt
+
+  # construct config.toml
+  touch config.toml
+  {
+    echo rpcAddr = \"http://localhost:26750\"
+    echo chainId = \"greenfield_9000-121\"
+  } > config.toml
+}
+
+######################
+# test create bucket #
+######################
+function test_create_bucket() {
+  set -e
+  cd ${workspace}/greenfield-cmd/build/
+  ./gnfd-cmd -c ./config.toml --home ./ sp ls
+  sleep 5
+  ./gnfd-cmd -c ./config.toml --home ./ bucket create gnfd://${BUCKET_NAME}
+  sleep 10
+}
+
+###########################################################
+# test upload and download file which size less than 16MB #
+###########################################################
+function test_file_size_less_than_16_mb() {
+  set -e
+  cd ${workspace}/greenfield-cmd/build/
+  ./gnfd-cmd -c ./config.toml --home ./ object put --contentType "application/json" ${workspace}/test/e2e/spworkflow/testdata/example.json gnfd://${BUCKET_NAME}
+  sleep 16
+  ./gnfd-cmd -c ./config.toml --home ./ object get gnfd://spe2etestbucket/example.json ./test_data.json
+  check_md5 ${workspace}/test/e2e/spworkflow/testdata/example.json ./test_data.json
+  cat test_data.json
+}
+
+##############################################################
+# test upload and download file which size greater than 16MB #
+##############################################################
+function test_file_size_greater_than_16_mb() {
+  set -e
+  cd ${workspace}/greenfield-cmd/build/
+  dd if=/dev/urandom of=./random_file bs=17M count=1
+  ./gnfd-cmd -c ./config.toml --home ./ object put --contentType "application/octet-stream" ./random_file gnfd://${BUCKET_NAME}/random_file
+  sleep 16
+  ./gnfd-cmd -c ./config.toml --home ./ object get gnfd://spe2etestbucket/random_file ./new_random_file
+  sleep 10
+  check_md5 ./random_file ./new_random_file
+}
+
+##################################
+# check two md5 whether is equal #
+##################################
+function check_md5() {
+  set -e
+  if [ $# != 2 ]; then
+    echo "failed to check md5 value; this function needs two args"
+    exit 1
+  fi
+  file1=$1
+  file2=$2
+  md5_1=$(md5sum ${file1} | cut -d ' ' -f 1)
+  md5_2=$(md5sum ${file2} | cut -d ' ' -f 1)
+  echo ${md5_1}
+  echo ${md5_2}
+
+  if [ "$md5_1" = "$md5_2" ]; then
+      echo "The md5 values are the same."
+  else
+      echo "The md5 values are different."
+      exit 1
+  fi
+}
+
+#######################
+# run sp workflow e2e #
+#######################
+function run_e2e() {
+  set -e
+  test_create_bucket
```

Comment:
No need to check the return value?

Reply:
I head bucket, download data and compare md5 to check data is correct.

Reply:
If there is an error, action will fail.